crates/chain/src/chain_data.rs

```diff
@@ -87,6 +87,7 @@ impl From<ChainPosition<ConfirmationTimeAnchor>> for ConfirmationTime {
 }
 
 /// A reference to a block in the canonical chain.
+/// When used as an anchor block, it is the same as the confirmation block.
```

Comment:
I prefer it written like this, with the `Anchor` stuff in a separate section.

```suggestion
///
/// `BlockId` implements [`Anchor`]. When a transaction is anchored to `BlockId`, the confirmation
/// block and anchor block are the same block.
```